### PR TITLE
Trim out unrealistic left and right turns. Find cases where multiple

### PR DESCRIPTION
**Pull request by dabreegster**
lanes all lead to a single lane via left/right turn, and just keep the
left/rightmost lane.

Sanity checked at Rainier X S Massachusetts, and 23rd X S Massachusetts.

Before:
before

After:
after

@michaelkirk 

Struggling through this PR really makes me want to unit test turn generation. But I'm still not sure how to keep the boilerplate of specifying input down. Maybe with enough helper functions, it'll just be something like specifying tags of incoming roads to one intersection, with some basic inferred geometry (just for turn type picking). Then a lightweight way to assert the set of turns generated.

### CODE REVIEW

File: map_model/src/make/turns.rs [hunk 1]
```diff
@@ -294,3 +297,67 @@ fn lc_penalty(t: &Turn, map: &Map) -> isize {
...
     ((from_idx as isize) - (to_idx as isize)).abs()
 }
+
+fn remove_merging_turns(map: &Map, input: Vec<Turn>, turn_type: TurnType) -> Vec<Turn> {
+    let mut turns = Vec::new();
+
+    // Group turns of the specified type by (from, to) road
+    let mut pairs: BTreeMap<(RoadID, RoadID), Vec<Turn>> = BTreeMap::new();
+    for t in input {
+        // Only do this for driving lanes
```
Comment (dabreegster) [inline map_model/src/make/turns.rs:307]: As the videos in the PR show, there's still funniness with turns leading to/from bike/bus lanes. Should handle that next, but hopefully only after some kind of unit testing is in place.

File: map_model/src/make/turns.rs [hunk 2]
```diff
@@ -294,3 +297,67 @@ fn lc_penalty(t: &Turn, map: &Map) -> isize {
...
+            continue;
+        }
+
+        // From one to many is fine
+        if group.iter().map(|t| t.id.src).collect::<HashSet<_>>().len() == 1 {
+            turns.extend(group);
+            continue;
+        }
+
+        // TODO Handle >1 left/right turn lane.
```
Comment (dabreegster) [inline map_model/src/make/turns.rs:335]: Another case to work on after unit tests (and finding a good example in OSM to validate).

File: map_model/src/objects/road.rs [hunk 3]
```diff
@@ -407,6 +407,17 @@ impl Road {
         result
     }
 
+    // lane must belong to this road. Offset 0 is the centermost lane on each side of a road, then
```
Comment (dabreegster) [inline map_model/src/objects/road.rs:410]: I used `offset` before and was so confused why it wasn't counting from the center. :(  So I restored this "deprecated" method. I still need to figure out how to name "offset" from center vs left side of the road better.

**Comment by dabreegster** [comment]
Oh yeah, still regenerating all map data and seeing if this helps get lakeslice running again without gridlock.

And https://github.com/dabreegster/abstreet/tree/osm_tr_v1 is an alternate solution to the original problem at Rainier / Massach, but it only partly helps, and is less clear.

**Comment by dabreegster** [comment]
Woops, uploaded new data, which means anyone pulling from HEAD right now is getting 404's. I'll merge this now, but would still like feedback. I'm going to attempt unit tests today.

File: map_model/src/make/turns.rs (hunk 2 above)
Comment (michaelkirk) [reply map_model/src/make/turns.rs:335]: possible test case:
https://www.openstreetmap.org/way/130308599

Southbound on stewart, turning left onto 2nd:
leftmost lane is turn only, second-from-left is straight *or* left.

roadway tagged like: `turn:lanes=left|left;through|none`

I'm not sure what's conventional though.

**Comment by michaelkirk** [comment]
Nice improvement!

File: map_model/src/make/turns.rs (hunk 2 above)
Comment (dabreegster) [reply map_model/src/make/turns.rs:335]: This one should be perfect, thanks!

**Pull request merged**